beacon_node/execution_layer/src/engine_api/http.rs

```diff
@@ -386,6 +384,81 @@ pub mod serde_logs_bloom {
     }
 }
 
+/// Serializes the `transactions` field of an `ExecutionPayload`.
+pub mod serde_transactions {
+    use super::*;
+    use eth2_serde_utils::hex;
+    use serde::ser::SerializeSeq;
+    use serde::{de, Deserializer, Serializer};
+    use std::marker::PhantomData;
+
+    type Value<T, N> = VariableList<Transaction<T>, N>;
+
+    #[derive(Default)]
+    pub struct ListOfBytesListVisitor<T, N> {
+        _phantom_t: PhantomData<T>,
+        _phantom_n: PhantomData<N>,
+    }
+
+    impl<'a, T: EthSpec, N: Unsigned> serde::de::Visitor<'a> for ListOfBytesListVisitor<T, N> {
+        type Value = Value<T, N>;
+
+        fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+            write!(formatter, "a list of 0x-prefixed byte lists")
+        }
+
+        fn visit_seq<A>(self, mut seq: A) -> Result<Self::Value, A::Error>
+        where
+            A: serde::de::SeqAccess<'a>,
+        {
+            let mut outer = VariableList::default();
+
+            while let Some(val) = seq.next_element::<String>()? {
+                let inner_vec = hex::decode(&val).map_err(de::Error::custom)?;
+                let opaque_transaction = VariableList::new(inner_vec).map_err(|e| {
+                    serde::de::Error::custom(format!("transaction too large: {:?}", e))
+                })?;
+                let transaction = Transaction::OpaqueTransaction(opaque_transaction);
+                outer.push(transaction).map_err(|e| {
+                    serde::de::Error::custom(format!("too many transactions: {:?}", e))
+                })?;
+            }
+
+            Ok(outer)
+        }
```

Comment:
I think we can avoid a handwritten visitor implementation by defining a `JsonTransaction` type that derives `(De)Serialize`, and then uses `serde` attributes to tweak the behaviour. This has the advantage of being declarative rather than imperative.

AFAICT you want the serialisation to match the `types::Transaction` enum, except without the `{"selector": 0, "value": value}` wrapping, which could be achieved by using the untagged enum representation `serde(untagged)`: https://serde.rs/enum-representations.html#untagged

Everything else could be reused from Transaction:

https://github.com/sigp/lighthouse/blob/f1eec453c45129ea4793536670aacdd6ee849370/consensus/types/src/execution_payload.rs#L8-L20

(and you don't need the `rename = "0"` hack because of the untagged repr)